### DISCUSSION TITLE
백엔드 레벨 4 해야 할 것

### DISCUSSION THREAD

**Discussion by jaehee329** [woowacourse-teams/2023-haru-study#452] [아이디어]
.

**Comment by aak2075** [comment]
부하테스트

**Reply by woosung1223** [reply]
수업 일정에 맞춰서 하기로 결정

**Comment by woosung1223** [comment]
프로바이더 추상화 -> 여러 종류의 Provider를 하나의 API로 제공한다.
(세션 빈 활용하면 어떨까?)

**Reply by woosung1223** [reply]
우선순위 하

**Comment by woosung1223** [comment]
로그아웃 블랙리스트 관리
(추후에 Redis로 관리하면 어떨까?)
[참고](https://velog.io/@boo105/Redis-%EB%A5%BC-%ED%86%B5%ED%95%9C-JWT-Blacklist-%EA%B5%AC%ED%98%84)

**Reply by woosung1223** [reply]
우선순위 중

**Comment by jaehee329** [comment]
EC2 한 대 추가 시 해야 할 일들
추가되는 EC2에 새로운 Production 서버와 dev를 띄운다
현재 production, dev, nginx가 떠 있는 서버에서는 dev를 뺀다.
docker compose를 활용하여 was와 함께 사용할 로깅&모니터링 툴을 함께 설치한다.

도커와 로드밸런싱 관련 공부를 해야 함.

**Reply by jaehee329** [reply]
최상

**Comment by jaehee329** [comment]
커멘드와 쿼리 분리

**Reply by jaehee329** [reply]
우선순위는 하

**Comment by jaehee329** [comment]
통계성 데이터를 schedule을 사용하여 새벽 시간에 주기적으로 수집한다.

데이터는 mysql의 db에 저장하고 추후 가공 방식을 결정한다.

**Reply by jaehee329** [reply]
우선순위는 낮으나 난이도는 어렵지 않아 누군가 맡아서 해도 됨

**Comment by woosung1223** [comment]
회원 탈퇴

**Reply by aak2075** [reply]
최하

**Comment by woosung1223** [comment]
oauth 게시 상태 변경
<img width="204" alt="image" src="https://github.com/woowacourse-teams/2023-haru-study/assets/78679830/e3cf1409-4b02-4b2a-bfd2-a4488815abb5">

**Reply by jaehee329** [reply]
최최하

**Comment by jaehee329** [comment]
DB 변경이 있는 배포를 새벽 시간에 예약하여 진행

**Reply by jaehee329** [reply]
상

**Reply by jaehee329** [reply]
해결 완료

**Comment by MoonJeWoong** [comment]
- 마코
  - room을 study로 바꾸기

- 테오
  - DB 스키마 변경 및 예약 배포 포함

- 모디
  - OSIV
  - readMe => 시스템 구성도 첨부해서 작성하기

- 히이로
  - google oauth 로직 트랜잭션 분리하기 
  - 트랜잭션 readOnly = true 설정
